### PR TITLE
feat: Remove transient data

### PR DESCRIPTION
**Pull request by freesig**

### CODE REVIEW

**Review by supiket — APPROVED** [review]
LGTM, I tried to link the issue for this but cannot find it 🤔

File: crates/constraint-vm/src/lib.rs [hunk 1]
```diff
@@ -266,9 +264,6 @@ pub fn step_op_access(
         }
         asm::Access::ThisPathway => access::this_pathway(access.solution.index, stack),
```
Comment (mohammadfawaz) [inline crates/constraint-vm/src/lib.rs:265]: `ThisPathway` should be removed too?

Comment (freesig) [reply crates/constraint-vm/src/lib.rs:265]: ah good point

**Review by mitchmindtree — APPROVED** [review]
This is AWESOME. I'll probably base #199 on this, happy to see it land :ok_hand:

Edit: Although do we want to merge this into `develop` instead?

**Comment by freesig** [comment]
Nah this goes into main. I think @mohammadfawaz needs it

**Pull request merged**